api/src/extensions/manager.ts

```diff
@@ -172,6 +188,20 @@ export class ExtensionManager {
 		if (this.options.watch && !wasWatcherInitialized) {
 			this.updateWatchedExtensions(this.extensions);
 		}
+
+		this.messenger.subscribe(this.extensionInstalledChannel, () => {
+			this.reload();
+		});
+	}
+
+	/**
+	 * installs an external extension from registry
+	 */
+
+	public async install(name: string, version?: string, registry?: string): Promise<void> {
+		await this.installationManager.install(name, version, registry);
+		//publish the message so that instances can reload extensions.
+		await this.messenger.publish(this.extensionInstalledChannel, {});
```

Comment:
after downloading the package from registry to configured location, publish a message so that all(including current) instance can reload the extensions async.